gengokit/generator/gen.go

```diff
@@ -62,32 +53,28 @@ func GenerateGokit(sd *svcdef.Svcdef, conf gengokit.Config) ([]truss.NamedReadWr
 
 // generateResponseFile contains logic to choose how to render a template file
 // based on path and if that file was generated previously. It accepts a
-// template path to render, a templateExecutor to apply to the template, and a
-// map of paths to files for the previous generation. It returns a
-// truss.NamedReadWriter representing the generated file
-func generateResponseFile(templFP string, te *gengokit.TemplateExecutor, prevGenMap map[string]io.Reader) (truss.NamedReadWriter, error) {
+// template path to render, a executor to apply to the template,
+// and . It returns a truss.NamedReadWriter representing the generated file.
+func generateResponseFile(executor *gengokit.Executor, file io.Reader, templFP string) (truss.NamedReadWriter, error) {
```

Comment:
This area is being refactored currently, but Zaq wanted middlewares in and so I have done this minimal refactor in order to get the middlewares in. 

So basically, I hear all of your points, and I have branches with the code refactored. But it is too large of a change to get into the repo in a reasonable time frame. So your points will be addressed in another refactor.